src/SofaCaribou/Mass/CaribouMass.inl

```diff
@@ -0,0 +1,506 @@
+        return Inherit1::canCreate(o, context, arg);
+    }
+
+    if (not requested_element_type.empty()) {
+        arg->logError("Requested element type is not '"+this_element_type+"'.");
+        return false;
+    }
+
+    std::string topology_path = arg->getAttribute("topology", "");
+    if (not topology_path.empty()) {
+        topology_path = topology_path.substr(1); // removes the "@"
+        // Make sure the specified topology has elements of type Element
+        auto topology = context->get<BaseObject>(topology_path);
+        auto caribou_topology = dynamic_cast<CaribouTopology *>(topology);
+        auto sofa_topology = dynamic_cast<sofa::core::topology::BaseMeshTopology *>(topology);
+        if (not caribou_topology and (not sofa_topology or not CaribouTopology::mesh_is_compatible(sofa_topology))) {
+            if (not sofa_topology) {
+                arg->logError("The path '" + topology_path + "' doesn't point to either a SOFA mesh topology, or a Caribou topology.");
+            } else {
+                arg->logError("Cannot deduce the element type from the specified SOFA topology '" + topology->getPathName() + "'.");
+            }
+            return false;
+        }
+    } else {
+        // Try to find a compatible topology in the current context
+        BaseObject * topology = nullptr;
+        auto objects = context->getObjects<BaseObject>(BaseContext::SearchDirection::Local);
+        for (auto * object : objects) {
+            auto caribou_topology = dynamic_cast<const CaribouTopology *>(object);
+            auto sofa_topology = dynamic_cast<const sofa::core::topology::BaseMeshTopology *>(object);
+            if (caribou_topology  or (sofa_topology and CaribouTopology::mesh_is_compatible(sofa_topology))) {
+                topology = object;
+                break;
+            }
+        }
+        if (not topology) {
+            arg->logError("Cannot find a topology in the current context from which the template '"+this_element_type+"' can be deduced.");
+            return false;
+        }
+
+        if (Inherit1::canCreate(o, context, arg)) {
+            arg->setAttribute("topology", "@" + topology->getPathName());
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    return Inherit1::canCreate(o, context, arg);
+}
+
+template<typename Element>
+void CaribouMass<Element>::assemble_mass_matrix() {
+    assemble_mass_matrix(*this->mstate->read (sofa::core::ConstVecCoordId::restPosition()));
+}
+
+template<typename Element>
+void CaribouMass<Element>::assemble_mass_matrix(const sofa::Data<VecCoord> & x0) {
+    using namespace sofa::core::objectmodel;
+
+    const sofa::helper::ReadAccessor<Data<VecCoord>> sofa_x0 = x0;
+    const auto nb_nodes = sofa_x0.size();
+    Eigen::Map<const Eigen::Matrix<Real, Eigen::Dynamic, Dimension, Eigen::RowMajor>> X (sofa_x0.ref().data()->data(),  nb_nodes, Dimension);
```

Comment:
Sport indeed!

This is a way to map SOFA's data structure with Eigen. In other words, `X` here will behave as if it was an Eigen matrix, for example with
```c++
const sofa::helper::ReadAccessor<Data<VecCoord>> sofa_x0 = x0;
Eigen::MatrixXd X (n, n);
copy(X, sofa_x0);
```

 but instead of allocating memory and doing a copy, it will directly use (map) the memory of SOFA's `Data<VecCoord>>`, hence no copy!